python-avd/pyavd/_cv/workflows/verify_inputs.py

```diff
@@ -29,47 +31,63 @@ def verify_device_inputs(devices: list[CVDevice], warnings: list[Exception], *,
     """
-    if (duplicated_devices := identify_duplicated_devices(devices)).detected():
+    if (duplicated_devices := identify_duplicated_devices(avd_devices)).detected():
         duplicated_devices_handler(
             duplicated_devices,
             warnings,
             strict_system_mac_address=strict_system_mac_address,
         )
 
 
-def identify_duplicated_devices(devices: list[CVDevice]) -> DuplicatedDevices:
+def identify_duplicated_devices(devices: list[AvdDevice]) -> DuplicatedDevices:
     """
-    Process list of CVDevice instances to identify those with overlapping serial_number or system_mac_address.
+    Process list of AvdDevice instances to identify those with overlapping serial_number or system_mac_address.
 
     Return DuplicatedDevices object containing:
-      - Information about CVDevices with overlapping serial_number.
-      - Information about CVDevices with overlapping system_mac_address where at least one of these devices has an unset `serial_number` value.
-      - Information about CVDevices with overlapping system_mac_address and set serial_number.
+      - Information about CVDevice with overlapping serial_number.
+      - Information about CVDevice with overlapping system_mac_address where at least one of these devices has an unset `serial_number` value.
+      - Information about CVDevice with overlapping system_mac_address and set serial_number.
     """
     duplicated_devices = DuplicatedDevices()
 
-    # Group devices based on <CVDevice>.serial_number as long as it's not None
+    if not devices:
+        return duplicated_devices
+
+    # we expect each device input to be represented only once as AvdDevice instance
+    LOGGER.debug("identify_duplicated_avd_devices() input devices: %s", len(devices))
+    cv_devices = [device.get_cv_device() for device in devices]
+
+    # Group devices based on <device>.serial_number as long as it's not None
     devices_grouped_by_serial_number = groupby_obj(
-        list_of_objects=[device for device in devices if device.serial_number is not None], attr="serial_number", skip_singles=True
+        list_of_objects=[device for device in cv_devices if device.serial_number is not None], attr="serial_number", skip_singles=True
     )
 
-    # Group devices based on <CVDevice>.system_mac_address as long as it's not None
+    # Group devices based on <device>.system_mac_address as long as it's not None
     devices_grouped_by_system_mac_address = groupby_obj(
-        list_of_objects=[device for device in devices if device.system_mac_address is not None], attr="system_mac_address", skip_singles=True
+        list_of_objects=[device for device in cv_devices if device.system_mac_address is not None], attr="system_mac_address", skip_singles=True
     )
 
-    # Populate list of CVDevice with duplicated serial_number values
+    # Populate list of AvdDevice with duplicated serial_number values
```

Comment:
This should be list of CVDevices, not AvdDevices

```suggestion
    # Populate list of CVDevice with duplicated serial_number values
```

Reply:
Done